lib/tasks/gettext.rake

```diff
@@ -35,6 +35,31 @@ namespace :gettext do
         end
    end
 
+   desc 'Update locale files with slightly changed English msgids using a csv file of old to new strings'
+   task :update_msgids_from_csv do
+       mapping_file = find_mapping_file(ENV['MAPPING_FILE'])
+       mappings = {}
+       File.foreach(mapping_file) do |csv_line|
+           from,to = CSV.parse(csv_line.gsub('\"', '""')).first
+           mappings[from] = to
+       end
+       Dir.glob("locale/**/app.po").each do |po_file|
+           lang_mappings = mappings.clone
+           lines = []
+           File.read(po_file).each_line do |line|
+
+               /^msgid "(.*)"/ =~ line
+               if $1 && mappings[$1]
+                    lines << "msgid \"#{lang_mappings.delete($1)}\""
+                 else
+                    lines << line
+                end
+           end
+           puts "Mappings unused: #{lang_mappings.keys}" unless lang_mappings.empty?
```

Comment:
These are mappings that were in `mappings.csv` that didn't get replaced in this file?

Should we print the filename that they didn't get used in?

```
Mappings unused in locale/cy/app.po:
  "Some mapping"
  "Another key"
```